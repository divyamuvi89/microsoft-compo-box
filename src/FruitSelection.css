body {
    font-family: Verdana, sans-serif; 
    margin: 0;
    padding: 0;
  }
  
  .fruit-selection-container {
    text-align: center;
    margin-top: 50px;
    display: flex;
    flex-direction: column;
    align-items: center;
    font-family: Verdana, sans-serif; 
  }
  
  .custom-dropdown {
    position: relative;
    display: inline-block;
    margin-top: 20px;
    width: 350px;
  }
  
  .fruit-input {
    padding: 10px;
    font-size: 16px;
    border: 1px solid #ccc;
    width: 350px;
    box-sizing: border-box;
    outline: none;
    transition: border-color 0.3s;
    font-family: Verdana, sans-serif; /* Apply Verdana font to the input */
    font-weight: bold; /* Make the text bold */
  }
  .fruit-input::after {
    content: "â–¼"; 
    position: absolute;
    top: 50%;
    right: 10px;
    transform: translateY(-50%);
    background-color: blue;
  }
  
  
  .fruit-input:focus {
    border-color: #3498db; /* Highlight border color on focus */
  }
  
  .dropdown-header {
    padding: 10px;
    background-color: #3498db;
    color: white;
    border-radius: 5px;
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
    width: 350px;
    text-align: center;
    font-weight: bold; 
  }
  
  .dropdown-list {
    list-style: none;
    max-height: 400px; 
    overflow-y: auto;
    padding: 0;
    margin: 0;
    position: absolute;
    background-color: #fff;
    border: 1px solid #3498db;
    width: 350px;
    border-radius: 5px;
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
  }
  
  .dropdown-list li {
    padding: 10px;
    transition: background-color 0.3s;
    font-family: Verdana, sans-serif; 
  }
  
  .dropdown-list li:hover {
    background-color: #f2f2f2;
  }
  
  .selected-fruit {
    font-weight: bold;
    margin-top: 10px;
  }
  