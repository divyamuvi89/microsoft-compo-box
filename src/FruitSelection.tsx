import React, {useState} from "react";
import './FruitSelection.css';

const fruits =['Apple', 'Orange', 'Banana', 'Grape','Kiwi','Mango', 'Pear','Pineapple' ];
const FruitSelection:React.FC = () => {
    const [selectedFruit, setselectedFruit] = useState<string>('');
    const [filteredFruits, setfilteredFruits] = useState<string[]>(fruits)
    const [showdropDown, setShowdropdown] =useState<boolean>(false);

    const handleChange = (e:React.ChangeEvent<HTMLInputElement>) => {
        const searchValue = e.target.value;
        setselectedFruit(searchValue);
        const filtered = fruits.filter(fruit => fruit.toLowerCase().includes(searchValue.toLowerCase()));
        setfilteredFruits(filtered);
    };
    const handleSelect = (fruit:string) =>{
        setselectedFruit(fruit);
        setShowdropdown(false);
        setfilteredFruits(fruits);
    }

    return(
        <div className="fruit-selection-container">
            <h2>Choose a fruit </h2>
            {selectedFruit && <p className="selected-fruit">Item Selected</p>}
           <div className="custom-dropdown">
            <input
            className="fruit-input"
            type = "text"
            placeholder="Search for a fruit...."
            list="fruits"
            value={selectedFruit}
            onClick={() => setShowdropdown(true)}
            onChange={handleChange}
            />
          {showdropDown && (
          <ul className="dropdown-list">
            {filteredFruits.map((fruit) => (
                <li key={fruit} onClick={() => handleSelect(fruit)}>{fruit}</li>
            ))}
          </ul>
          )}
          </div>
            </div>
           
        
    );
};
export default FruitSelection;