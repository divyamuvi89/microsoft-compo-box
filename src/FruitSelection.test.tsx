import React from "react";
import renderer from "react-test-renderer";
import FruitSelection from "./FruitSelection"; // Update the path to your component file

test("renders FruitSelection component correctly", () => {
  const tree = renderer.create(<FruitSelection />).toJSON();
  expect(tree).toMatchSnapshot();
});
