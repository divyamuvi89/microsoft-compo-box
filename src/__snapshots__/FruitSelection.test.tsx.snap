// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders FruitSelection component correctly 1`] = `
<div
  className="fruit-selection-container"
>
  <h2>
    Choose a fruit 
  </h2>
  <div
    className="custom-dropdown"
  >
    <input
      className="fruit-input"
      list="fruits"
      onChange={[Function]}
      onClick={[Function]}
      placeholder="Search for a fruit...."
      type="text"
      value=""
    />
  </div>
</div>
`;
